tests/integration/safehtml/safehtml_test.go

```diff
@@ -1,70 +1,79 @@
 package safehtml_test
 
 import (
-	"io"
-	"strings"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"github.com/google/go-safeweb/safehttp"
 	"github.com/google/safehtml"
 	"github.com/google/safehtml/template"
 )
 
-type dispatcher struct {
-	wr io.Writer
-}
+type dispatcher struct{}
 
-func (d *dispatcher) Write(resp safehttp.Response) error {
+func (d *dispatcher) Write(rw http.ResponseWriter, resp safehttp.Response) error {
 	switch x := resp.(type) {
 	case safehtml.HTML:
-		_, err := d.wr.Write([]byte(x.String()))
+		_, err := rw.Write([]byte(x.String()))
 		return err
 	default:
 		panic("not a safe response type")
 	}
 }
 
-func (d *dispatcher) ExecuteTemplate(t safehttp.Template, data interface{}) error {
+func (d *dispatcher) ExecuteTemplate(rw http.ResponseWriter, t safehttp.Template, data interface{}) error {
 	switch x := t.(type) {
 	case *template.Template:
-		return x.Execute(d.wr, data)
+		return x.Execute(rw, data)
 	default:
 		panic("not a safe response type")
 	}
 }
 
 func TestHandleRequestWrite(t *testing.T) {
-	b := &strings.Builder{}
-	d := &dispatcher{wr: b}
+	m := safehttp.NewMachinery(func(rw safehttp.ResponseWriter, _ *safehttp.IncomingRequest) safehttp.Result {
+		return rw.Write(safehtml.HTMLEscaped("<h1>Escaped, so not really a heading</h1>"))
+	}, &dispatcher{})
```

Comment:
Don't @ me, but I am starting to think the dispatcher should somehow be registered in the machinery and not passed every time a user of the package needs to write a response.

Reply:
It is registered in the machinery. `NewMachinery` is a constructor here.

Reply:
I misread the parentheses, disregard my comment pls.